### PR TITLE
fix failed tests on Windows #8737

### PR DESCRIPTION
**Pull request by jddxf**

### CODE REVIEW

File: scripts/jest/ts-preprocessor.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ function compile(content, contentFilename) {
       // the file path into backslashes on Windows.
       filename = path.normalize(filename);
       var reactRegex = new RegExp(
-        path.join('/', '(?:React|ReactDOM)(?:\.d)?\.ts$')
+        '\\' + path.join('/', '(?:React|ReactDOM)(?:\.d)?\.ts$')
```
Comment (gaearon) [inline scripts/jest/ts-preprocessor.js]: Can you explain why this helps exactly? There is a comment right above:

```js
 
       // `path.normalize` and `path.join` are used to turn forward slashes in
        // the file path into backslashes on Windows.
```

Are they not helping? Why?

Ideally I’d like to see us use one separator style, and normalize paths when necessary.

Comment (jddxf) [inline scripts/jest/ts-preprocessor.js]: They help to normalize the path.But we need more work to construct a `RegExp` instance. 

image

The backslash added here is to escape the leading separator, which is an special character in regular expression.

Oh,Wait.All of a sudden, I find the dots need to be escaped too.So I'll change it to:

```
var reactRegex = new RegExp(
  path.join('/', '(?:React|ReactDOM)(?:\.d)?\.ts$')
    .replace(/[\\.]/g, '\\$&')
);
```
Or more simple, if we don't mind matching files like XXXReact.d.ts:

```
var reactRegex = /(?:React|ReactDOM)(?:\.d)?\.ts$/;
```
@gaearon Which do you think would be better?

**Comment by gaearon** [comment]
Thank you for fixing this!

**Comment by john-brunelle** [comment]
Test Comment

**Pull request merged**